### PR TITLE
fix: validate config and secret sets against the schema

### LINKED ISSUE #1972: Reenable secret validation.
Steps to repro:
- Add a module that defines a secret
- Set the secret using `ftl secret set ...`

We get this error:
> ftl: error: unknown: declaration "secretname" not found

Looks like we are looking up the active schema before setting a secret and erroring if it is not found. And a schema is only active if it was successfully deployed with runners. But when creating a secret that has not been set yet it can not deploy.

### PR DESCRIPTION
**Pull request by safeer**
Fixes #1972 by re-reverting #1978, passing validation for decl's that aren't found, and adding integration tests around validation pre- and post-deploy.

### CODE REVIEW

File: common/projectconfig/integration_test.go [hunk 1]
```diff
@@ -59,3 +59,37 @@ func TestFindConfig(t *testing.T) {
...
 }
+
+func TestConfigValidation(t *testing.T) {
+	in.Run(t, "./validateconfig/ftl-project.toml",
+		in.CopyModule("validateconfig"),
+
+		// Global sets never error.
+		in.Chdir("validateconfig", in.Exec("ftl", "config", "set", "key", "--inline", "valueTwo")),
+		in.Chdir("validateconfig", in.Exec("ftl", "config", "set", "key", "--inline", "2")),
+		in.ExecWithExpectedOutput("\"2\"\n", "ftl", "config", "get", "key"),
```
Comment (deniseli) [inline common/projectconfig/integration_test.go:70]: nit: backticks avoid the need to escape the double quotes inside the string (here and below) 
```
`"string"\n`
```

File: common/projectconfig/integration_test.go [hunk 2]
```diff
@@ -59,3 +59,37 @@ func TestFindConfig(t *testing.T) {
 		checkConfig("."),
 	)
 }
+
+func TestConfigValidation(t *testing.T) {
```
Comment (deniseli) [inline common/projectconfig/integration_test.go:63]: nice this is awesome

**Review by deniseli — APPROVED** [review]
I only reviewed the portions that were different from https://github.com/TBD54566975/ftl/pull/1858. Looks great!

File: common/projectconfig/integration_test.go (hunk 1 above)
Comment (safeer) [reply common/projectconfig/integration_test.go:70]: can't use escape characters with backticks unfortunately, so that `\n` would be two chars

Comment (deniseli) [reply common/projectconfig/integration_test.go:70]: ahhhh drats

**Pull request merged**